keras_tuner/engine/hyperparameters.py

```diff
@@ -565,6 +578,42 @@ def name_scope(self, name):
         finally:
             self._name_scopes.pop()
 
+    def explore_space(self, hypermodel):
```

Comment:
Perhaps "compute_search_space" or something along these lines would be more precise, what do you think? We want to avoid confusion between "explore" and "search"

In addition, perhaps this should be a method on the `Hypermodel` base class, rather than the `Hyperparameters` class. Hyperparameters are lower in the abstraction hierarchy and thus should not be aware of hypermodels. What you are "exploring" is really the search space implicitly defined by the hypermodel (so it's a property of the hypermodel).

e.g.:

```python
class HyperModel:
    def compute_search_space(self, hp):
       ...
       return hp
```

Reply:
Yes, "compute_search_space" is a better name.

Agree that it shouldn't be implemented in the `HyperParameters` class, but I think it makes more sense to put it in the `Oracle` class.

`HyperModel` is a search space, which is stateless, just defining how to use the hyperparameters to create a model. `HyperParameter` is the state to pass to the search space to instantiate one model. The `Oracle` controls the interaction between them, like generating pass a new state to instantiate a new model from the search space. Therefore, if we want to compute the search space through multiple instantiations from the search space using different states, it should be done on the level of the `Oracle`.

The new implementation would be `HyperParameters` only record the active and inactive conditions for the current run, just like recording the used `HyperParameters`. The `Oracle` would collect the condtions like collecting `HyperParameter`s.

Reply:
Sounds good.